tasks/init.rb

```diff
@@ -5,7 +5,18 @@
 require 'open3'
 
 def get(fact)
-  cmd_string = "facter -p #{fact}"
+  if ENV.has_key? 'Path' then
```

Comment:
What was here originally worked on Windows. I wouldn't change how Windows operates, only hardcode the path on everything else.

Reply:
Are you referring to the original version prior to any of my commits?  If so, I am running into the same path related failure with just a base vmpooler win-2012r2-x86 image running facter_task against it.  It is only with this update that it works.

Reply:
Using a base 2017.3.1 install and a newly installed Windows 2012r2 x86_64 image from the pooler, it fails with the current master branch and works with the update:

```
# git status
# On branch master
nothing to commit (working directory clean)
# puppet task run facter_task fact=osfamily -n jwjnmd901tjjqj4.delivery.puppetlabs.net
Starting job ...
New job ID: 18
Nodes: 1

Started on jwjnmd901tjjqj4.delivery.puppetlabs.net ...
Failed on jwjnmd901tjjqj4.delivery.puppetlabs.net
  Error: Task finished with exit-code 1
  error :
  status : failure

Job failed. 1 node failed, 0 nodes skipped, 0 nodes succeeded.
Duration: 3 sec
# git checkout fm-6472-use-ruby-facter
Branch fm-6472-use-ruby-facter set up to track remote branch fm-6472-use-ruby-facter from origin.
Switched to a new branch 'fm-6472-use-ruby-facter'
# puppet task run facter_task fact=osfamily -n jwjnmd901tjjqj4.delivery.puppetlabs.net
Starting job ...
New job ID: 19
Nodes: 1

Started on jwjnmd901tjjqj4.delivery.puppetlabs.net ...
Finished on node jwjnmd901tjjqj4.delivery.puppetlabs.net
  status : windows

Job completed. 1/1 nodes succeeded.
Duration: 4 sec
```

Reply:
Hmm, I don't understand that yet. Although thinking about it more, it'd probably need something like what you've added to support Bolt. A better implementation would look up the install location in the registry.

Reply:
Looking into why it's failing... it appears to be successfully running facter, but then getting a failure at the end that results in a non-zero exit code.

Reply:
My guess was that it was trying to run the incorrect facter.  We want to run `C:\Program Files\Puppet Labs\Puppet\bin\facter.bat`, but the path puts `C:/Program Files/Puppet Labs/Puppet/facter/lib/../bin` ahead of it:

```
Path : C:/Program Files/Puppet Labs/Puppet/facter/lib/../bin;C:/Program Files/Puppet Labs/Puppet/facter/lib/../../puppet/bin;C:\Program Files\Puppet Labs\Puppet\pxp-agent\bin;C:\Program Files\Puppet Labs\Puppet\puppet\bin;C:\Program Files\Puppet Labs\Puppet\facter\bin;C:\Program Files\Puppet Labs\Puppet\sys\ruby\bin;C:\ProgramData\Boxstarter;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files\Git\cmd;C:\Packer\SysInternals;C:\Program Files\Puppet Labs\Puppet\bin
```

Calling that binary directly without the facter.bat wrapper appears to result in an error.

Reply:
Ah, could be. Although I think the paths are there, because stdout has the fact result in my run (added some debugging to the task).